cfgov/unprocessed/apps/filing-instruction-guide/js/fig-sidenav.js

```diff
@@ -1,68 +1,121 @@
-/* eslint-disable complexity, no-undefined */
-/* ==========================================================================
-   Scripts for Report Sidenav organism
-   ========================================================================== */
-
-const main = document.querySelector( 'main' );
-const sidenav = main.querySelector( '.o-report-sidenav' );
-const tocHeaders = sidenav.querySelectorAll( '.m-nav-link' );
-const headerOffset = main.offsetTop - 10;
-const headers = main.querySelectorAll( 'h2.report-header, h3.report-header' );
-
-const offsets = [];
-const primaryOffsets = [];
-let lastTargetIndex;
-
-// Initialize offsets for calculating what to highlight
-( function() {
-  for ( let i = 0; i < headers.length; i++ ) {
-    offsets.push( headers[i].offsetTop + headerOffset );
-    primaryOffsets.push( headers[i].tagName === 'H2' );
-  }
-} )();
+import Expandable from '@cfpb/cfpb-expandables/src/Expandable';
+import { DESKTOP, viewportIsIn } from '../../../js/modules/util/breakpoint-state.js';
+
+const appRoot = () => document.querySelector( 'main.o-fig' );
+
+const getNavItem = target => appRoot().querySelector( `.m-nav-link[href="${ target }"]` );
```

Comment:
I wonder if it would be worth it here & for `getNavItemContainer` to pre-calculate these results on load and save them into two `Map`s an `itemMap` and a `containerMap` with the `target` nodes as keys. Then you just `itemMap.get(target)` in the event handler, which is much faster than a live dom query. When you build the maps you could probably do a single DOM query then just do attribute checks which would be quite fast.

This doesn't seem janky in the current iteration, but I'm always twitchy with scroll/intersection.